lib/debugger/ui.luau

```diff
@@ -136,65 +150,128 @@ local function ui(debugger, loop)
+						font = Enum.Font.Gotham,
+					})
+					plasma.space(5)
+
+					local listOfSystems = {}
+
+					for _, system in systems do
+						local samples = loop.profiling[system]
+						if samples then
+							local duration = rollingAverage.getAverage(samples)
+							durations[system] = duration
+							longestDuration = math.max(longestDuration, duration)
+						end
 					end
 
-					table.insert(items, {
-						text = systemName(system),
-						sideText = averageFrameTime,
-						selected = debugger.debugSystem == system,
-						system = system,
-						icon = icon,
-						barWidth = duration / longestDuration,
-						index = index,
-					})
-				end
+					for index, system in systems do
+						local nameOfSystem = systemName(system)
+
+						if searchResults[eventName][nameOfSystem] == nil then
+							continue
+						end
+
+						local averageFrameTime = ""
+						local icon
+
+						local duration = durations[system] or 0
+						local humanDuration, unit = formatDuration(duration)
+						averageFrameTime = string.format("%.0f%s", humanDuration, unit)
+
+						if duration > 0.004 then -- 4ms
+							icon = "\xe2\x9a\xa0\xef\xb8\x8f"
+						end
+
+						if loop._systemErrors[system] then
+							icon = "\xf0\x9f\x92\xa5"
+						end
+
+						local barWidth
+						if longestDuration == 0 then
+							barWidth = 0
+						else
+							barWidth = duration / longestDuration
+						end
+
+						local systemIsDisabled = loop._skipSystems[system]
+						local systemName = systemName(system)
+						local length = string.len(systemName)
+
+						if systemIsDisabled then
+							length += 4
+						end
+
+						table.insert(listOfSystems, {
+							text = if systemIsDisabled then `<i><s>{systemName}</s></i>` else systemName,
+							sideText = if systemIsDisabled then `<b>{"(disabled)"}</b>` else averageFrameTime,
+							selected = debugger.debugSystem == system,
+							system = system,
+							icon = icon,
+							barWidth = barWidth,
+							index = index,
+						})
+					end
+
+					local systemList = custom.selectionList(listOfSystems, custom)
+					local selected = systemList:selected()
+					local rightClicked = systemList:rightClicked()
+					hoveredSystem = systemList:hovered()
+					if selected then
+						if selected.system == debugger.debugSystem then
+							debugger.debugSystem = nil
+						else
+							debugger.debugSystem = selected.system
+						end
+					elseif rightClicked and rightClicked.system then
+						local current = loop._skipSystems[rightClicked.system]
 
-				local selected = custom.selectionList(items):selected()
-				if selected then
-					if selected.system == debugger.debugSystem then
-						debugger.debugSystem = nil
+						if current == nil then
+							loop._skipSystems[rightClicked.system] = true
+						else
+							loop._skipSystems[rightClicked.system] = not current
+						end
+					end
+
+					if debugger.debugSystem then
+						debugger.debugSystemRuntime = durations[debugger.debugSystem]
```

Comment:
```lua
local debugSystem
if selected then 
    local selectedSystem = selected.system
    debugSystem = debugger.debugSystem
    if selectedSystem == debugSystem then 
        debugSystem = nil
        debugSystem.system = debugSystem
    else
        debugSystem = selectedSystem
    end
elseif rightClicked then
    local rightClickedSystem = rightClicked.system
    if rightClickedSystem then 
        -- put skipSystems anywhere reachable to this codeblock honestly since you reuse it in other callpoints.
        local skipSystems = loop._skipSystems
        -- if it is nil, `not nil` will turn to true!
        skipSystems[rightClickedSystem] = not skipSystems[rightClickedSystem]
    end
end
if debugSystem then 
  debugger.debugSystemRuntime = durations[debugSystem]
end

```